src/solvers/refinement/string_refinement.cpp

```diff
@@ -886,71 +901,89 @@ static std::string string_of_array(const array_exprt &arr)
 /// solver to constant expressions given by the current model
 void debug_model(
-  const replace_mapt &symbol_resolve,
+  const union_find_replacet &symbol_resolve,
   messaget::mstreamt &stream,
   const namespacet &ns,
   const std::size_t max_string_length,
   const std::function<exprt(const exprt &)> &super_get,
   const std::vector<symbol_exprt> &boolean_symbols,
   const std::vector<symbol_exprt> &index_symbols)
 {
+  const auto &eom=messaget::eom;
   const std::string indent("  ");
-  for(auto it : symbol_resolve)
+  std::set<array_string_exprt> char_array_in_axioms;
+#if 0
+  generator.debug_arrays_of_pointers(stream);
+#endif
+  for(auto pair : symbol_resolve.to_vector())
+    char_array_in_axioms.insert(to_array_string_expr(pair.first));
+
+  for(auto  arr : char_array_in_axioms)
   {
-    if(const auto refined=expr_cast<string_exprt>(it.second))
-    {
-      stream << "- " << from_expr(ns, "", to_symbol_expr(it.first)) << ":\n"
-             << indent << indent << "in_map: "
-             << from_expr(ns, "", *refined) << '\n'
-             << indent << indent << "resolved: "
-             << from_expr(ns, "", *refined) << '\n';
-      const exprt &econtent=refined->content();
-      const exprt &elength=refined->length();
-
-      exprt len=super_get(elength);
-      len=simplify_expr(len, ns);
-      const exprt arr=get_array(
-        super_get,
-        ns,
-        max_string_length,
-        econtent, len);
-      if(arr.id()==ID_array)
-        stream << indent << indent << "as_string: \""
-               << string_of_array(to_array_expr(arr)) << "\"\n";
-      else
-        stream << indent << indent << "as_char_array: "
-               << from_expr(ns, "", arr) << "\n";
+    const exprt model=
+      get_char_array_in_model(super_get, ns, max_string_length, stream, arr);
 
-      stream << indent << indent << "size: " << from_expr(ns, "", len) << '\n';
-    }
-    else
-    {
-      INVARIANT(
-        is_char_array(ns, it.second.type()),
-        string_refinement_invariantt("symbol_resolve should only map to "
-          "refined_strings or to char_arrays, and refined_strings are already "
-          "handled"));
-      exprt arr=it.second;
-      replace_expr(symbol_resolve, arr);
-      stream << "- " << from_expr(ns, "", to_symbol_expr(it.first)) << ":\n";
-      stream << indent << indent << "resolved: "
-              << from_expr(ns, "", arr) << "\n";
-      exprt arr_model=get_array(super_get, arr);
-      stream << indent << indent << "char_array: "
-             << from_expr(ns, "", arr_model) << '\n';
-    }
+    stream << "- " << from_expr(ns, "", arr) << ": "
+           << from_expr(ns, "", model) << eom;
   }
 
-  for(const auto &it : boolean_symbols)
+  for(const auto it : boolean_symbols)
   {
       stream << " - " << it.get_identifier() << ": "
              << from_expr(ns, "", super_get(it)) << '\n';
   }
 
-  for(const auto &it : index_symbols)
+  for(const auto it : index_symbols)
```

Comment:
Why remove the `&`? Also, probably use `sym` or something, since `it` is not an iterator.